Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -2452,6 +2452,23 @@ bool AccessibilityObject::ignoredFromModalPresence() const
     if (modalNode->document().frame() != this->frame())
         return false;
     
+    // Some objects might be outside of a modal, but are linked to elements inside of it. Don't ignore those.
+    for (RefPtr ancestor = this; ancestor; ancestor = ancestor->parentObject()) {
+        for (auto& controller : ancestor->controllers()) {
+            if (RefPtr controllerObject = dynamicDowncast<AccessibilityObject>(controller)) {
+                if (controllerObject->isModalDescendant(*modalNode))
+                    return false;
+            }
+        }
+
+        for (auto& activeDescendant : ancestor->activeDescendantOfObjects()) {
+            if (RefPtr activeDescendantObject = dynamicDowncast<AccessibilityObject>(activeDescendant)) {
```

Comment:
I think we should just downcast<AccessibilityObject>. This should never be an isolated object